spec/features/users_feature_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+feature "User can sign in and out" do
+  context "user not signed in and on the homepage" do
+    it "should see a 'sign in' link and a 'sign up' link" do
+      visit('/')
+      expect(page).to have_link('Sign in')
+      expect(page).to have_link('Sign up')
+    end
+
+    it "should not see 'sign out' link" do
+      visit('/')
+      expect(page).not_to have_link('Sign out')
+    end
+  end
+
+  context "user signed in on the homepage" do
+    before do
+      visit('/')
+      click_link('Sign up')
+      fill_in('Email', with: 'test@example.com')
```

Comment:
Have a look at this for how to refactor this sort of thing: https://gist.github.com/ptolemybarnes/2dfda607b85d01e113b0